### ISSUE TITLE
Add ramp option to ingest config

### ISSUE THREAD

**Issue by j6k4m8** [jhuapl-boss/ingest-client#37]
# Issue 

Many services from which our plugins pull have a gradual rate-limit increase. Especially because these ingest jobs can saturate that limit very quickly (say, by running with a high `-p`arallelism count), it'd be beneficial to support a gradual ramp-up for use alongside the `-p` option.

# Proposed solution 1: CLI argument

Pass a `--ramp` `-r` flag that contains the number of seconds in between the addition of subsequent workers:

```shell
boss-ingest -p 36 --ramp 10s my-config.json
```

The above command will run 36 workers in parallel, adding another after every 10 seconds (after 30 seconds, three workers will be running. At 6 minutes, all workers will be running).

One could imagine adding support for other string formats like `--ramp 5m` or `--ramp 1h`; or perhaps adding support for a _count_ of workers to be added at each increment, such as `--ramp "10,5m"`. But this starts getting complicated and may be better served by:

# Proposed solution 2: JSON config

I like this solution slightly less because it would potentially require a change to the ingest-client json schema, but it also enables much more flexibility with designing the ramp-up curve shape.

For example, borrowing from the [k6](https://docs.k6.io/docs/options) options:

```js
{
    stages: [
        { time: "10s", workers: 10 },
        { time: "20s", workers: 10 },
        { time: "20m", workers: 20 },
        { time: "40m", workers: 20 }
    ]
}
```

The above configuration would add 10 workers at the 0:00:10 mark, another 10 at the 0:00:20 mark, and then twenty workers each at the 0:20:00 and 0:40:00 minute marks, for a total of 60 workers.

Such a configuration would be particularly useful for, say, pulling from Google Cloud Storage, where [read-rate limits are doubled no sooner than 20 minutes after an ingest starts](https://cloud.google.com/storage/docs/request-rate#auto-scaling).

**Comment by j6k4m8** [comment]
Revisiting this... I can't believe how unintelligible my proposed command-line args are. `--ramp 6h -p 30` should ramp linearly from 0 to 30 workers over 6 hours.

**Comment by j6k4m8** [comment]
fixed by #50
